### PR TITLE
Running commands remotely: On restrictive webhost

### PR DESCRIPTION
**Pull request by porg**
- Added a 4th method which works on restrictive webhosts too:
  - Remote non-interactive shell resolves aliases and runs wp as alias with php.
- Transformed the other 3 methods from free text to sections with a headline.
  - All headlines created during this edit now also produce nice HTML anchor links intended as permalinks - Internal jumplinks, which I already make use of. - And to guide externals visitors directly to their recommended method (chat, email, social media)
  - I tried to give the headlines a timeless and long lasting name as good as possible.
    - Please treat them as permalinks.
    - If wrong, then please correct them now, before they start their lifecycle as permalinks/deeplinks.
- It took me 1.5 working days of research and trial and error until I finally found a way to run wp-cli remotely on my webhost.
  - Eventually was glad when I found a way. And I want to share this!
  - Also put quite some effort into this writeup.
  - I hope that this makes it to the website.
  - Please feel free to further correct / amend. You are the expert(s)!

### CODE REVIEW

File: running-commands-remotely.md [hunk 1]
```diff
@@ -112,7 +112,8 @@ wp_users:user_email
 
 Running a command remotely through SSH requires having `wp` accessible on the `$PATH` on the remote server. Because SSH connections don’t load `~/.bashrc` or `~/.zshrc`, you may need to specify a custom `$PATH` when using `wp --ssh=<host>`.  A few ways to make `wp` available on the remote server are:
 
-+ **Copy WP-CLI binary to `$HOME/bin`**:  
+### Copy WP-CLI binary to $HOME / bin
```
Comment (danielbachhuber) [inline running-commands-remotely.md:115]:
```suggestion
### Copy WP-CLI binary to $HOME/bin
```

File: running-commands-remotely.md [hunk 2]
```diff
@@ -129,10 +130,46 @@ If `$HOME/bin` is not already in your path, then you can define it in your `~/.b
 PATH="$HOME/bin:$PATH"
 ```
 
-+ **Specify the `$PATH` in  `~/.ssh/environment`**:  
+### Specify the $PATH in $HOME / .ssh / environment
```
Comment (danielbachhuber) [inline running-commands-remotely.md:133]:
```suggestion
### Specify the $PATH in $HOME/.ssh/environment
```

**Review by danielbachhuber — APPROVED** [review]
This is great, @porg — thank you!

**Comment by porg** [comment]
## Please restore the extra spaces!
- Yes these path style as in the headlines may seem a bit strange, but this is necessary for the anchor links, which serve as permalinks both for arriving and also internally as cross-links.
- The real correct paths are then anyhow in the free text.

## Apart from this
- When will this go live on the website?

**Comment by danielbachhuber** [comment]
@porg Sorry about that! Want to fix them in a follow-up PR?

I forget how often this syncs, but I think it's at least once a day.

**Comment by porg** [comment]
- I lost my git practise and would take long for this or worse screw up. I kindly ask you to please do this reverting.

- Alternatively: If you know a solution to define a Headline and define the anchor link for it manually (in the Markdown file) which then makes it to the live HTML site, this would ofc be the "best of both worlds".
  - But on the other side, future editors may not be familiar with this special markup and discard it.
  - The current compromise is quite acceptable I'd say.
  
I am so proud that this my contribution makes it online 🙂

**Comment by danielbachhuber** [comment]
@porg Sounds good! I've reverted the change: https://github.com/wp-cli/handbook/commit/b401f12e7e7556965b46056fc10eba27a1bfde8f

**Pull request merged**